lib/internal/Magento/Framework/DB/Adapter/Pdo/Mysql.php

```diff
@@ -2233,10 +2233,12 @@ public function createTemporaryTable(\Magento\Framework\DB\Ddl\Table $table)
      */
     public function createTemporaryTableLike($temporaryTableName, $originTableName, $ifNotExists = false)
     {
-        $ifNotExistsSql = ($ifNotExists ? 'IF NOT EXISTS' : '');
+        $ifNotExistsSql = ($ifNotExists ? ' IF NOT EXISTS' : '');
         $temporaryTable = $this->quoteIdentifier($this->_getTableName($temporaryTableName));
         $originTable = $this->quoteIdentifier($this->_getTableName($originTableName));
-        $sql = sprintf('CREATE TEMPORARY TABLE %s %s LIKE %s', $ifNotExistsSql, $temporaryTable, $originTable);
+        $originCreate = $this->fetchPairs(sprintf('SHOW CREATE TABLE %s', $originTable));
```

Comment:
```suggestion
        $originCreate = $this->fetchOne("SHOW CREATE TABLE {$originTable}");
```
You can use fetchOne to get result
Also sprintf is unnecessary

Reply:
@kandy Unfortunately I can't use `fetchOne` because `SHOW CREATE TABLE` returns two fields. First is table name, the second is SQL query.

I like you suggestion about `sprintf`, though :+1:.